EIP-4844 Experimental Support

rauljordan opened the pull request:
This PR adds some experimental support for 4844 txs by using the latest commit of https://github.com/mdehoog/go-ethereum

MariusVanDerWijden commented on transactions.go:
```diff
@@ -161,6 +175,39 @@ func new1559Tx(nonce uint64, to *common.Address, gasLimit uint64, chainID, tip,
+			Gas:                 view.Uint64View(gasLimit),
+			To:                  types.AddressOptionalSSZ{Address: (*types.AddressSSZ)(to)},
+			Value:               view.Uint256View(*val),
+			BlobVersionedHashes: versionedHashes,
```
Could you also add some data pls? I am going to modify fuzzyVM (which creates the tx data) to also create calls to the blob precompile as well as the datahash opcode

MariusVanDerWijden commented on transactions.go:
```diff
@@ -130,6 +132,18 @@ func RandomValidTx(rpc *rpc.Client, f *filler.Filler, sender common.Address, non
+	case 10:
+		// 4844 transaction with AL
+		tx := types.NewTransaction(nonce, to, value, gas, gasPrice, code)
+		al, err := CreateAccessList(rpc, tx, sender)
```
Would be nice to also have a version without al, since some clients still don't support creating an al, so they would not send blob txs

MariusVanDerWijden reviewed (commented):
Two small nits, looks very good thank you for your contribution

0xGabi commented on transactions.go:
```diff
@@ -161,6 +175,39 @@ func new1559Tx(nonce uint64, to *common.Address, gasLimit uint64, chainID, tip,
+	}
+	val, ok := uint256.FromBig(value)
+	if !ok {
+		panic("fee cap not big int")
```
```suggestion
		panic("value not big int")
```

0xGabi commented on transactions.go:
```diff
@@ -161,6 +175,39 @@ func new1559Tx(nonce uint64, to *common.Address, gasLimit uint64, chainID, tip,
+		panic(err)
+	}
+	txData := types.SignedBlobTx{
+		Message: types.BlobTxMessage{
```
Comparing with https://github.com/Inphi/blob-utils/blob/master/main.go#L125-L136 I think we are missing `ChainID` and  `MaxFeePerDataGas`.

0xGabi reviewed (changes requested):
Thank you for adding 4844 support. I left a small review.

rauljordan commented:
Hi @MariusVanDerWijden and @0xGabi thanks for the review. I have added the requested fields and random data. One question I had was how big should the max data size be per transaction for the purpose of this PR. I know the total we are aiming for in a single block is definitely < 1Mb, probably closer to 256-512Kb. I added a randomBlobData func which generates data to be encoded into a blob of random size from 0 to 512Kb. Let me know if this is unreasonable, as we might be creating a single tx with 512Kb worth of data being encoded into a blob (is that even allowed)?

MariusVanDerWijden commented:
Hey I fucked up a bit with git. I force pushed some weird old stuff onto your branch which overwrote all your changes.
I rescued your commit to another branch now

MariusVanDerWijden commented:
It's here: https://github.com/MariusVanDerWijden/tx-fuzz/pull/8
The problem is that I can't reopen your PR :( but your commit history with attribution is preserved on the other branch as well.
Again sorry about that

rauljordan commented:
@MariusVanDerWijden ha no worries about it - it's happened to me as well 😆. Was happy to make this small contribution to save you some time
